x-pack/plugins/apm/server/lib/rum_client/ui_filters/get_es_filter.ts

```diff
@@ -23,5 +25,16 @@ export function getEsFilter(uiFilters: UIFilters) {
       };
     }) as ESFilter[];
 
+  if (
+    uiFilters.environment &&
+    uiFilters.environment !== ENVIRONMENT_ALL.value
+  ) {
+    mappedFilters.push({
+      term: {
+        [SERVICE_ENVIRONMENT]: uiFilters.environment,
+      },
+    });
+  }
```

Comment:
Perhaps use `environmentQuery`:

```suggestion
   return [...mappedFilters, ...environmentQuery(environment)]
```